core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -220,11 +228,28 @@ object ConsoleProducer {
     options = tryParse(parser, args)
 
     CommandLineUtils.printHelpAndExitIfNeeded(this, "This tool helps to read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
+
+    if (!options.has(bootstrapServerOpt) && !options.has(brokerListOpt))
+      CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, bootstrapServerOpt)
+
+    if (options.has(bootstrapServerOpt))
+      CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, bootstrapServerOpt)
+
+    if (options.has(brokerListOpt))
+      CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
+
 
     val topic = options.valueOf(topicOpt)
+
+    val bootstrapServer = options.valueOf(bootstrapServerOpt)
```

Comment:
Probably want to do something like this:

```
val brokerHostsAndPorts = options.valueOf(if (options.has(bootstrapServerOpt)) bootstrapServerOpt else brokerListOpt)
ToolsUtils.validatePortOrDie(parser, brokerHostsAndPorts)
```

Reply:
Updated with suggestion